### PR TITLE
use Casbin RBAC model at server side

### PR DESCRIPTION
**Pull request by kingiw**
Signed-off-by: kingiw <kingiw@hotmail.com>

Temporarily, Casbin doesn't support RBAC+ABAC model (wait for the merge of  https://github.com/casbin/casbin/pull/583/commits/7c9f78f6a964183442201037e723600080df7a8c), and I simply implement the RBAC part for enforce if a user is root moderator or node moderator by now. 

The policies are inserted into our database.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,13 +7,19 @@ require (
 	github.com/aliyun/aliyun-oss-go-sdk v2.1.4+incompatible
 	github.com/aliyun/aliyun-sts-go-sdk v0.0.0-20171106034748-98d3903a2309
 	github.com/astaxie/beego v1.12.1
+	github.com/casbin/casbin v1.7.0
```
Comment (hsluoyz) [inline go.mod]: Why 1.7.0?

File: object/file.go [hunk 2]
```diff
@@ -14,6 +14,10 @@
 
 package object
 
+import (
```
Comment (hsluoyz) [inline object/file.go]: no bracket for one row.

File: go.mod [hunk 3]
```diff
@@ -7,13 +7,18 @@ require (
 	github.com/aliyun/aliyun-oss-go-sdk v2.1.4+incompatible
 	github.com/aliyun/aliyun-sts-go-sdk v0.0.0-20171106034748-98d3903a2309
 	github.com/astaxie/beego v1.12.1
+	github.com/casbin/casbin/v2 v2.10.0
+	github.com/casbin/xorm-adapter v1.0.0 // indirect
```
Comment (hsluoyz) [inline go.mod]: Why is `xorm-adapter v1.0.0` still used?

File: go.sum [hunk 4]
```diff
@@ -23,9 +25,18 @@ github.com/beego/goyaml2 v0.0.0-20130207012346-5545475820dd/go.mod h1:1b+Y/CofkY
 github.com/beego/x2j v0.0.0-20131220205130-a0352aadc542/go.mod h1:kSeGC/p1AbBiEp5kat81+DSQrZenVBZXklMLaELspWU=
 github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973/go.mod h1:Dwedo/Wpr24TaqPxmxbtue+5NUziq4I4S80YR8gNf3Q=
 github.com/bradfitz/gomemcache v0.0.0-20180710155616-bc664df96737/go.mod h1:PmM6Mmwb0LSuEubjR8N7PtNe1KxZLtOUHtbeikc5h60=
+github.com/casbin/casbin v1.7.0 h1:PuzlE8w0JBg/DhIqnkF1Dewf3z+qmUZMVN07PonvVUQ=
```
Comment (hsluoyz) [inline go.sum]: Still Casbin 1.7.0

Comment (hsluoyz) [inline go.sum]: Still Casbin 1.7.0

File: go.sum [hunk 5]
```diff
@@ -23,9 +25,18 @@ github.com/beego/goyaml2 v0.0.0-20130207012346-5545475820dd/go.mod h1:1b+Y/CofkY
...
 github.com/casbin/casbin v1.7.0/go.mod h1:c67qKN6Oum3UF5Q1+BByfFxkwKvhwW57ITjqwtzR1KE=
+github.com/casbin/casbin/v2 v2.0.1/go.mod h1:YcPU1XXisHhLzuxH9coDNf2FbKpjGlbCg3n9yuLkIJQ=
+github.com/casbin/casbin/v2 v2.10.0 h1:hBGwn5dnOiA2agPwpedz0a3ke/OVnWksyoHm7dW/bWc=
+github.com/casbin/casbin/v2 v2.10.0/go.mod h1:XXtYGrs/0zlOsJMeRteEdVi/FsB0ph7KgNfjoCoJUD8=
+github.com/casbin/xorm-adapter v1.0.0 h1:AJq8+6BhmjD6Oajth8UUpuE/2RAX3PR+s/T98+4tQr8=
```
Comment (hsluoyz) [inline go.sum]: Same again.

**Comment by hsluoyz** [comment]
@kocoler please review.

File: go.sum (hunk 4 above)
Comment (hsluoyz) [inline go.sum]: Still Casbin 1.7.0

File: object/file.go [hunk 6]
```diff
@@ -14,6 +14,9 @@
 
 package object
 
+import . "github.com/casbin/casbin-forum/authz"
+
+
```
Comment (kocoler) [inline object/file.go]: I think there should be only one blank line.

File: authz/util.go [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package authz
+
+import (
```
Comment (kocoler) [inline authz/util.go]: I think there should be no parentheses.

**Comment by kocoler** [comment]
I just found some problems that conflict with my understanding of format.

**Pull request merged**